### PR TITLE
[CI] Update Convergence Criteria to Fix Flaky KMeans Test

### PR DESCRIPTION
**Pull request by tarang-jain**
Addresses the KMeans flaky test in https://github.com/rapidsai/cuvs/issues/2100. Since we bit match centroids, we limit the number of iterations to ensure that both the OOC and in-memory run the same number of iters. The flaky test is due to numerical differences arising out of atomic operations (perhaps due to different ordering or adding several batch sums in the OOC case versus a single atomic op in the in-memory case) -- `raft::reduce_rows_by_key`. 

UPDATES 08/25/2026:
Rather than clamping the number of iters, we fix the convergence criteria.
rtxpro6000:
main: 9/50 runs were failing.
After this change: 100/100 runs pass.

### CODE REVIEW

File: cpp/tests/cluster/kmeans.cu [hunk 1]
```diff
@@ -381,6 +381,9 @@ class KmeansFitBatchedTest : public ::testing::TestWithParam<KmeansBatchedInputs
     params.tol                 = testparams.tol;
     params.rng_state.seed      = 1;
     params.oversampling_factor = 0;
+    // Limit the number of iterations to ensure same number of iterations for reference and batched
+    // code paths.
+    params.max_iter = 3;
```
Comment (cjnolet) [inline cpp/tests/cluster/kmeans.cu]: This fix doesn't seem particularly robust, and appears to me like it'll reduce the coverage for the tests, right? Default max_iters for k-means is much higher. How do we test in a way where we can validate its ability to stop early without having to reduce the max-iters to the point where we are essentially not testing the early stopping at all?

Comment (tarang-jain) [reply cpp/tests/cluster/kmeans.cu]: I think the point of the test when we are bit matching centroids is to ensure that the accumulation of partial sums in the OOC setting in a single iteration matches the in-memory setting. The test fails when there is an edge case -- the variation due to floating point precision in the arithmetic is just enough to push the difference in centroids between consecutive iterations to fall below the convergence criteria -- in-memory converges one iteration sooner in than OOC.

**Review by cjnolet — APPROVED** [review]
